src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -198,4 +212,55 @@ public String getDeterministHashFromFile(String chainTaskId) {
                 chainTaskId, sign.getR(), sign.getS(), sign.getV());
         return Optional.of(sign);
     }
+
+    public boolean encryptResult(String chainTaskId) {
+        String beneficiarySecretFilePath = smsService.getBeneficiarySecretFilePath(chainTaskId);
+        String resultZipFilePath = getResultZipFilePath(chainTaskId);
+        String taskOutputDir = configurationService.getTaskOutputDir(chainTaskId);
+
+        if (!new File(beneficiarySecretFilePath).exists()) {
+            log.info("No beneficiary secret file found, will continue without encrypting result [chainTaskId:{}]", chainTaskId);
```

Comment:
The name of the method here is encryptResult but we may skip the encryption. Name and behavior of the method do not match, on or the other should be changed then (maybe introduce a isEncryptionNeeded method?)

Reply:
Totally true yeah! I split it.